core/domain/exp_domain.py

```diff
@@ -2499,3 +2499,541 @@ def add_contribution_by_user(self, contributor_id):
+                                    for property in
+                                    self.PROPERTIES_RELATED_TO_SOLUTION) and
+                                    change.property_name not in
+                                    changed_translations[state_name] and
+                                    STATE_PROPERTY_INTERACTION_SOLUTION not in
+                                    changed_properties[state_name]):
+                                change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_SOLICIT_ANSWER_DETAILS):
+                    if state_name in changed_properties:
+                        if (old_exp_states.interaction.id ==
+                                current_exp_states.interaction.id and
+                                old_exp_states.solicit_answer_details ==
+                                current_exp_states.solicit_answer_details):
+                            change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_CARD_IS_CHECKPOINT):
+                    change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_RECORDED_VOICEOVERS):
+                    if state_name in changed_properties:
+                        if (all(property not in
+                                changed_properties[state_name]
+                                for property in
+                                self.PROPERTIES_RELATED_TO_VOICEOVERS) and
+                                STATE_PROPERTY_RECORDED_VOICEOVERS not in
+                                changed_properties[state_name]):
+                            change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+            elif change.cmd == CMD_ADD_WRITTEN_TRANSLATION:
+                state_name = change.state_name
+                if change.state_name in state_names_of_renamed_states:
+                    state_name = (
+                        state_names_of_renamed_states[change.state_name])
+                if state_name in old_to_new_state_names:
+                    # Here we will send the changelist, frontend_version,
+                    # backend_version and exploration to the admin, so
+                    # that the changes related to state renames can be
+                    # reviewed and the proper conditions can be written
+                    # to handle those cases.
+                    return ({
+                        'changes_are_mergeable': False,
+                        'send_email': True
+                    })
+                if state_name not in changed_translations:
+                    changed_translations[state_name] = []
```

Comment:
Can we add a `continue` here?

Reply:
No, because it may be the case when translation is not changed in that state but content containing property is changed. So we can not skip this case.